### ISSUE TITLE
Anomalies and Hovmoeller plots break with multiple countries

### ISSUE THREAD

**Issue by luigibrancati** [ECMWFCode4Earth/atmospheric-explorer#21]
I selected multiple countries (Italy, Spain, France and Portugal) on the Streamlit app and tried generating the Anomalies and Hovmoeller plots, but they broke. The issue is with the clipping of the xarray dataset most likely.

**Comment by elisaliv** [comment]
@GiuliaFiantanese and I started investigating this bug tonight.
We tried to replicate your error selecting the same countries listed in the issue:
* _Anomalies_ is not working, and the error looks related to clipping as you speculated:
![image](https://github.com/ECMWFCode4Earth/atmospheric-explorer/assets/49102376/2fa7b5ea-57ac-4414-a658-b1a46a244259)
* _Latitude Hovmoeller_ is working, but it does not involve clipping anyway. We should make it clearer that it's not related to the selected countries.
* _Pressure level Hovmoeller_ is working:
![image](https://github.com/ECMWFCode4Earth/atmospheric-explorer/assets/49102376/d05ef1b9-90ba-473a-857f-a7822f4cdd68)
However, it only works when selecting multiple levels. We should add a note that you cannot create a pressure level Hovmoeller plot with only one level.
* _Model level Hovmoeller_ is not implemented.

So, these are the next steps to solve this issue:
- [ ] Debug the error for the _anomalies_ plot, by calling the function on a Jupyter notebook
- [ ] Add a note or re-think the UI for the _latitude Hovmoeller_ plot, to make it clear that it's a global plot and unrelated to selected countries
- [ ] Add a note for the _pressure level Hovmoeller_ plot, and/or force in some way selecting multiple pressure levels. This kind of plot does not make sense for a single level anyway
- [ ] [Out of scope for this issue] Implement _model level Hovmoeller_ or consider removing it if it's not needed

**Comment by elisaliv** [comment]
We further investigated the issue. The problem seems to be with Portugal data, as we had already noticed at the beginning.
If we try to create the anomalies plot with multiple countries that don't include Portugal, we do not get any error.

In the clipping function (`clip_and_concat_countries`, line 24 with `rio.clip`), if we have `drop=True` as argument we get this error:
```
NoDataInBounds: No data found in bounds. Data variable: tcno2
```

If we use `drop=False`, the function works but there seem to be no data in the anomalies plot (for Portugal only):
![image](https://github.com/ECMWFCode4Earth/atmospheric-explorer/assets/49102376/e3176470-c7e0-46f7-a172-98b12f5ac332)

We also tried to create a geospatial plot out of the same data, which supports this idea:
![image](https://github.com/ECMWFCode4Earth/atmospheric-explorer/assets/49102376/9dcbf32f-999d-41ee-9f56-1fe6fef0d614)

Strangely enough, we don't have such issues if we select Vatican as a country. So it's probably not a matter of small shapefile, but something else.

Our suggested solution would be to leave `drop=False` as default. This way, plots wouldn't break anyway.
We should also discuss with mentors about this issue.
